production/ksonnet/loki/shipper.libsonnet

```
@@ -24,21 +24,23 @@
     compactor_pvc_class: 'fast',
     index_period_hours: if self.using_shipper_store then 24 else super.index_period_hours,
     loki+: if self.using_shipper_store then {
-      storage_config+: {
+      storage_config+: if $._config.using_boltdb_shipper then {
         boltdb_shipper+: {
           shared_store: $._config.boltdb_shipper_shared_store,
           active_index_directory: '/data/index',
           cache_location: '/data/boltdb-cache',
         },
+      } else {} + if $._config.using_tsdb_shipper then {
```

Does `else if` work here?

```suggestion
      } else if $._config.using_tsdb_shipper then {
```

---

I think we'll want to have both configured if they are being used in different periods. `else if` would limit us to using a single config

---

Right